src/main/java/org/zone/region/flag/entity/player/display/title/TitleMessageDisplay.java

```diff
@@ -3,40 +3,121 @@
 import net.kyori.adventure.text.Component;
 import net.kyori.adventure.title.Title;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 import org.spongepowered.api.entity.living.player.Player;
+import org.zone.ZonePlugin;
+import org.zone.config.node.ZoneNodes;
 import org.zone.region.flag.entity.player.display.MessageDisplay;
 import org.zone.region.flag.entity.player.display.MessageDisplayType;
 import org.zone.region.flag.entity.player.display.MessageDisplayTypes;
 
+import java.time.Duration;
+import java.util.Optional;
+
 public class TitleMessageDisplay implements MessageDisplay {
 
     private @NotNull Component subTitle;
-    private @NotNull Title.Times times;
+    private @Nullable Duration fadeIn;
+    private @Nullable Duration stay;
+    private @Nullable Duration fadeOut;
 
-    public TitleMessageDisplay(@NotNull Component subTitle, @NotNull Title.Times times) {
-        this.subTitle = subTitle;
-        this.times = times;
+    public TitleMessageDisplay(@NotNull TitleMessageDisplayBuilder builder) {
+        this.subTitle = builder.getSubTitle();
+        this.fadeIn = builder.getFadeIn();
+        this.stay = builder.getStay();
+        this.fadeOut = builder.getFadeOut();
     }
 
+    /**
+     * Gets the subtitle of the Title Message
+     *
+     * @return The subtitle
+     */
     public @NotNull Component getSubTitle() {
         return this.subTitle;
     }
 
-    public @NotNull Title.Times getTimes() {
-        return this.times;
+    /**
+     * Gets the fade in duration of the Title message
+     *
+     * @return Optional nullable version of fade in duration
+     */
+    public @NotNull Optional<Duration> getFadeIn() {
+        return Optional.ofNullable(this.fadeIn);
+    }
+
+    /**
+     * Gets the stay duration of the Title message
+     *
+     * @return Optional nullable version of stay duration
+     */
+    public @NotNull Optional<Duration> getStay() {
+        return Optional.ofNullable(this.stay);
+    }
+
+    /**
+     * Gets the fade out duration of the Title message
+     *
+     * @return Optional nullable version of fade out duration
+     */
+    public @NotNull Optional<Duration> getFadeOut() {
+        return Optional.ofNullable(this.fadeOut);
```

Comment:
Personally I would have it return the default value if it is null, instead of using a Optional

Reply:
ah `@Nullable`

Reply:
Nope. You know i dont like returning null. 

Read my first comment again

Reply:
um default value.